pkg/engine/mutate/patch/strategicPreprocessing.go

```diff
@@ -163,7 +163,21 @@ func processListOfMaps(logger logr.Logger, pattern, resource *yaml.RNode) error
 					}
 				}
 			}
+			if resource == nil {
```

Comment:
Why `resource` is nil here?

Reply:
In this scenario and especially in CLI, the resource does not have the volumes field defined. In preprocessing the pattern related to the volume field, the resource does not contain volume field and this functions returns nil error instead of returning the global condition error. This in turn will skip the rule processing.

Reply:
I see, thanks!

If the resource node is nil, `preProcessRecursive` always returns nil error, do we still need to invoke it?

Reply:
We need to invoke, so that it will recurse again to check if the pattern is matching and as the resource is empty it will throw the appropriate error in other function.

If not invoked, the global anchor does not have matching resource and we are returning nil error which leads to preProcessing condition to be successful. Resource will be mutated even if the global anchor condition is not matched.

Reply:
Looked into the code, it returns typed errors on condition failures. Makes sense!